components/PlantDetails.js

```diff
@@ -40,12 +48,17 @@ export default function PlantDetails() {
   if (!plant) {
     return <h1>No Plant found</h1>;
   }
-
+  const isMyPlant = isMyPlantfunction(plant);
   const handleBackButton = () => {
     router.back();
```

Comment:
I'm sorry - this is from an older PR: router.back() will not work when you start on a detailspage (e.g. when you get the url from a friend). You should better use the next Link component instead of button.